tests/framework-tests/core/js/DataBindingTests.js

```diff
@@ -903,23 +903,51 @@ fluid.tests.fluid6442.checkIt = function (component) {
     }
 };
 
+fluid.defaults("fluid.tests.fluid6442options", {
+    listeners: {
+        "onResourcesLoaded.checkIt": {
+            func: "fluid.tests.fluid6442.checkIt",
+            args: "{that}"
+        },
+        "onResourceError.failTest": function (err) {
+            jqUnit.fail("Failure fetching resource: " + err);
+        }
+    }
+});
+
 jqUnit.asyncTest("FLUID-6442: Modelised relocalisation should be accessible via resources", function () {
```

Comment:
I have a question for this test:

1. If the model driven localization occurs at the startup with the initial model value `model.resourceLoader.locale === "en"`, the fetched message value of `component.resources.messages.parsed.courses` should be from `message1_en.json` file and equals to "These courses will require a lot of grading".
2. At line 924, when a model change on `resourceLoader.locale` is triggered, the `component.resources.messages.parsed.courses` value should be fetched from `message1_en_ZA.json` and equals to "These courses will require a lot of marking".

It seems the resource fetch at the startup doesn't happen for this test. I wonder why?

Reply:
It doesn't happen because the model update occurs synchronously, before the I/O can resolve - as the comment reads, "Change the locale before the queued I/O can resolve. In addition this tests promise cancellation propagation.". Given the I/O is in progress, it is cancelled in favour of the new model update and so only one fetch is seen.

Reply:
Got it. Thanks for the explanation.